parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -0,0 +1,318 @@
+        return filename + (
+            ".txt" if (len(filename) < 4 or filename[-4:] != ".txt") else ""
+        )
+
+    def file_iter(self, filename, start=0):
+        """generator to get the file line by line"""
+        with open(filename, "r", encoding="utf8") as file:
+            for line_number, line in enumerate(file):
+                if line_number < start:
+                    continue
+                # sanitizing
+                # remove any space characters at end of line
+                line = line.rstrip()
+                # replace ’ (typographique quote) to simple quote '
+                line = line.replace("’", "'")
+                # replace commas that have no space around by a lower comma character
+                # and do the same for escaped comma (preceded by a \)
+                # (to distinguish them from commas acting as tags separators)
+                line = re.sub(r"(\d),(\d)", r"\1‚\2", line)
+                line = re.sub(r"\\,", "\\‚", line)
+                # removes parenthesis for roman numeral
+                line = re.sub(r"\(([ivx]+)\)", r"\1", line, flags=re.I)
+                yield line_number, line
+        yield line_number, ""  # to end the last entry if not ended
+
+    def normalizing(self, line, lang="default"):
+        """normalize a string depending of the language code lang"""
+        line = unicodedata.normalize("NFC", line)
+
+        # removing accent
+        if lang in ["fr", "ca", "es", "it", "nl", "pt", "sk", "en"]:
+            line = re.sub(r"[¢£¤¥§©ª®°²³µ¶¹º¼½¾×‰€™]", "-", line)
+            line = unidecode.unidecode(line)
+
+        # lower case except if language in list
+        if lang not in []:
+            line = line.lower()
+
+        # changing unwanted character to "-"
+        line = re.sub(r"[\u0000-\u0027\u200b]", "-", line)
+        line = re.sub(r"&\w+;", "-", line)
+        line = re.sub(
+            r"[\s!\"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]",
+            "-",
+            line,
+        )
+
+        # removing excess "-"
+        line = re.sub(r"-+", "-", line)
+        line = line.strip("-")
+        return line
+
+    def remove_stopwords(self, lc, words):
+        """to remove the stopwords that were read at the beginning of the file"""
+        if lc in self.stopwords:
+            words_to_remove = self.stopwords[lc]
+            new_words = []
+            for word in words.split("-"):
+                if word not in words_to_remove:
+                    new_words.append(word)
+            return ("-").join(new_words)
+        else:
+            return words
+
+    def add_line(self, line):
+        """to get a normalized string but keeping the language code "lc:" , used for id and parent tag"""
+        lang = line[:2]
+        new_line = lang + ":"
+        new_line += self.remove_stopwords(lang, self.normalizing(line, lang))
```

Comment:
@BryanH01, I think this should be `line[3:]` since it doesn't build the `is_child_of` relationship without it.
```suggestion
        new_line += self.remove_stopwords(lang, self.normalizing(line[3:], lang))
```